### PR TITLE
Avoid messing the git repo

### LINKED ISSUE #566: assets/files/etc/chrony.conf modified during deploy
**Describe the bug**
Since we merged https://github.com/openshift-metal3/dev-scripts/pull/479 we leave the dev-scripts tree dirty after a deployment, because `utils.sh` does this:

```
echo "pool ${ntp} iburst" >> assets/files/etc/chrony.conf
```

After a deployment `assets/files/etc/chrony.conf` ends up in a modified state that needs a git reset - instead we should template this without modifying the git tracked files.

Also I think if you run multiple deployments without a git reset, we'll append the servers multiple times to the dirty `assets/files/etc/chrony.conf` ?

### PR DESCRIPTION
**Pull request by e-minguez**
Fixes #566

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/690/

**Comment by e-minguez** [comment]
Can we trigger the CI job again @derekhiggins ? I'm not sure if the failure is because the PR :$ Thanks!

File: utils.sh [hunk 1]
```diff
@@ -44,6 +45,14 @@ function create_cluster() {
...
     cp ${assets_dir}/install-config.yaml{.tmp,}
     $OPENSHIFT_INSTALLER --dir "${assets_dir}" --log-level=debug create cluster
+
+    # Clean up
+    [ -f assets/files/etc/chrony.conf.tmp ] && \
+      mv assets/files/etc/chrony.conf{.tmp,}
+
+    for file in assets/templates/{99_worker-chronyd-custom.yaml,99_master-chronyd-custom.yaml}; do
+      [ -f ${file} ] && rm ${file}
+    done
```
Comment (hardys) [inline utils.sh]: These are already cleaned up in `ocp_cleanup.sh` but wouldn't it be easier/better to stop messing with the assets/files completey, and instead modify/copy things into assets/generated (e.g after generate_assets runs, which already cleans that directory, and it's already ignored by git)

**Comment by hardys** [comment]
> Can we trigger the CI job again @derekhiggins ? I'm not sure if the failure is because the PR :$ Thanks!

You can re-trigger by copying the parameters into http://10.8.144.11:8080/job/dev-tools/build?delay=0sec but note I made some suggestions on how we might handle this more cleanly via the existing generated assets dir.

File: utils.sh (hunk 1 above)
Comment (e-minguez) [reply utils.sh]: As `chrony.conf` needs to be modified after the yaml files are templated, I opted of doing it 'manually' instead.
Let me know about this solution.
Thanks @hardys !

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/704/

**Comment by e-minguez** [comment]
My bad. I believe I've fixed.

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/705/

**Comment by hardys** [comment]
Thanks, this looks better, also tested locally although not with custom NTP servers.

**Pull request merged**